app/services/punches_service.rb

```diff
@@ -0,0 +1,15 @@
+class PunchesService
+  def self.create(punches, user)
+    deletes = punches.map { |punch| punch['from'] }
```

Comment:
You could call this variable something like `dates_to_delete` or `punch_dates_to_delete` for better readability.

Reply:
I will do it